### PR TITLE
Fix Searchkick functionality, update listings controller, listing mod…

### PR DESCRIPTION
**Pull request by JustineH**
…el, neighbourhood model, and CSVs

### CODE REVIEW

File: app/controllers/listings_controller.rb [hunk 1]
```diff
@@ -4,20 +4,18 @@ class ListingsController < ApplicationController
...
-
     @listings = Listing.search(params[:query] + "*", misspellings: {edit_distance: false}, where: params[:query_options].deep_symbolize_keys)
 
-    # Listing.search("Killarn", autocomplete: true, limit: 10)
-
-     @user = current_user.id
-
-    @bookmarked_listings = current_user.bookmarked_listings
+    if current_user
+      @user = current_user.id
```
Comment (AaronBetterOfficeApps) [inline app/controllers/listings_controller.rb:10]: This seem pointless, could you not just use `current_user` helper in your view?

File: app/controllers/listings_controller.rb [hunk 2]
```diff
@@ -4,20 +4,18 @@ class ListingsController < ApplicationController
...
+      @bookmarked_listings = current_user.bookmarked_listings
```
Comment (AaronBetterOfficeApps) [inline app/controllers/listings_controller.rb:11]: Same as above

File: app/controllers/listings_controller.rb [hunk 3]
```diff
@@ -4,20 +4,18 @@ class ListingsController < ApplicationController
...
+    end
     # @bookmarked_listing = Listing.find(Bookmark.where(user_id: current_user.id).first.listing_id)
     # @test = current_user.bookmarked_listings.create(listing_id: params[:listing_id])
   end
 
   def show
-    @neighbourhoods = Neighbourhood.all
+ 
   end
```
Comment (AaronBetterOfficeApps) [inline app/controllers/listings_controller.rb:19]: Is this action supposed to show a specific neighbourhood?

File: app/controllers/listings_controller.rb [hunk 4]
```diff
@@ -34,7 +32,7 @@ def create
     # @listing.user = current_user
     @listing = current_user.listings.build(listing_params)
     if @listing.save
-      redirect_to @listing, notice: 'Listing was successfully entered'
+      redirect_to @listing, notice: 'Listing was successfully entered.'
```
Comment (AaronBetterOfficeApps) [inline app/controllers/listings_controller.rb:35]: 👍

File: app/models/listing.rb [hunk 5]
```diff
@@ -1,14 +1,22 @@
 class Listing < ActiveRecord::Base
-  searchkick synonyms: [["Kitsilano", "Kits"], ["Avenue", "Ave", "Av", "Ave.", "Av."], ["Street", "St", "St."], ["East", "E"], ["West", "W"]]
-            # autocomplete: ['neighbourhood_id'] 
+  searchkick synonyms: [["kitsilano", "kits"], ["avenue", "ave", "av", "ave.", "av."], ["street", "st", "st."], ["east", "e"], ["west", "w"]]
 
   belongs_to :user
   belongs_to :neighbourhood
 
   has_attached_file :image, :styles => { :medium => "300x300>", :thumb => "100x100>" } 
   # validates_attachment :image, content_type: { content_type: ["image/jpg", "image/jpeg", "image/png", "image/gif"] }
```
Comment (AaronBetterOfficeApps) [inline app/models/listing.rb]: remove unused code

File: app/models/listing.rb [hunk 6]
```diff
@@ -1,14 +1,22 @@
...
+      neighbourhood: neighbourhood.name,
+      bedrooms: bedrooms,
+      bathrooms: bathrooms,
+      price: price,
+      home_type: home_type
+    }
+  end
 
   # validates :address, presence: true, length: { minimum: 10}
   # validates :unit_number, numericality: true
```
Comment (AaronBetterOfficeApps) [inline app/models/listing.rb:21]: remove un-used code

File: db/seeds.rb [hunk 7]
```diff
@@ -46,22 +45,18 @@ def townhouse_address_factory
   @townhouse_addresses.delete(add)
   add
 end
+
   # neighbourhoods = ["Downtown Vancouver", "West End", "Yaletown", "Coal Harbour", "Gastown", "Chinatown", "Downtown Eastside", "False Creek", "Kitsilano", "Marpole", "Point Grey", "Commercial Drive", "Mount Pleasant", "Oakridge", "Kerrisdale", "Arbutus Ridge", "Dunbar-Southlands", "False Creek", "Granville Island/Fairview", "Shaughnessy", "South Cambie", "Grandview-Woodland", "Hastings-Sunrise", "Killarney", "Strathcona", "Riley Park-Little Mountain", "Sunset", "Victoria-Fairview", "Dunbar"]
 
   # neighbourhoods.each do |neighbourhood|
   #   Neighbourhood.create(name:neighbourhood)
   # end
```
Comment (AaronBetterOfficeApps) [inline db/seeds.rb:53]: remove un-used code

File: app/models/listing.rb [hunk 8]
```diff
@@ -1,11 +1,21 @@
...
+  def search_data
+    {
+      address: address,
+      neighbourhood: neighbourhood.name,
+      bedrooms: bedrooms,
+      bathrooms: bathrooms,
+      price: price,
+      home_type: home_type
+    }
+  end
```
Comment (AaronBetterOfficeApps) [inline app/models/listing.rb:17]: This looks like a fantastic Idea 💯

File: app/controllers/listings_controller.rb (hunk 3 above)
Comment (JustineH) [inline app/controllers/listings_controller.rb:19]: No, the set_listing method now selects which params we want to show.

**Pull request merged**